client/driver/mock_driver.go

```diff
@@ -78,14 +77,30 @@ type MockDriverConfig struct {
 // MockDriver is a driver which is used for testing purposes
 type MockDriver struct {
 	DriverContext
-	fingerprint.StaticFingerprinter
+
+	// isShutdown is an internal concept to use to track whether the driver
+	// should be shut down
+	isShutdown bool
 
 	cleanupFailNum int
 }
 
 // NewMockDriver is a factory method which returns a new Mock Driver
 func NewMockDriver(ctx *DriverContext) Driver {
-	return &MockDriver{DriverContext: *ctx}
+	md := &MockDriver{DriverContext: *ctx}
+
+	// if the shutdown configuration options are set, start the timer here.
+	// This config option defaults to false
+	if ctx.config != nil && ctx.config.ReadBoolDefault(fingerprint.ShutdownPeriodicAfter, false) {
+		duration, err := ctx.config.ReadInt(fingerprint.ShutdownPeriodicDuration)
```

Comment:
How about storing the duration and start time as a field on the mock driver and changing the Fingerprint method to be `if time.Now().After(d.shutdownFingerprintTime) {`. Avoids the other function and goroutine